pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -213,7 +213,6 @@ func (d *defaultDriver) Reconcile(ctx context.Context) *reconciler.Results {
 			msg := "Could not update remote clusters in Elasticsearch settings"
 			d.ReconcileState.AddEvent(corev1.EventTypeWarning, events.EventReasonUnexpected, msg)
 			log.Error(err, msg, "namespace", d.ES.Namespace, "es_name", d.ES.Name)
-			results.WithResult(defaultRequeue)
```

Comment:
Doesn't this change the behaviour for every other error case? Eg. es client or k8s client temporary errors?

Reply:
I missed the if below. So actually, the difference in behaviour would only be there if we return `false` requeue and no error - I believe this can happen if `remoteClustersToDelete` is `0` and `updateSettings(esClient, remoteClustersToApply)` fails in `updateSettingsInternal`.

Reply:
The goal is to not change the current behaviour and just avoid to requeue twice. I also missed the edge case. It should be good if we also always requeue on error: 7c319fb.